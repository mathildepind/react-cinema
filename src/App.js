import React from 'react';
import SearchForm from './SearchForm';
import Result from './Result';
import MovieDetails from './MovieDetails';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      movies : [],
      movieData : {},
      movie : null
    };

    this.handleClickReceiver = this.handleClickReceiver.bind(this);
    this.handleMovieClickReceiver = this.handleMovieClickReceiver.bind(this);
    this.fetchMovie = this.fetchMovie.bind(this);
  }

  handleClickReceiver(input){
    fetch(`http://www.omdbapi.com/?s=${input}&apikey=25a585bd`)
    .then((response)=>{
     return response.json()
    }).then((myJsonData)=>{
     let movieArray = myJsonData.Search;
     this.setState({movies : movieArray});

    }).catch((error) =>{
     console.log(error);
    });
  }

  handleMovieClickReceiver(input){
    let encodedInput = encodeURIComponent(input);
    fetch(`http://www.omdbapi.com/?t=${encodedInput}&apikey=25a585bd`)
    .then((response)=>{
     return response.json()
    }).then((myJsonData)=>{
     let movieObject = myJsonData;
     this.setState({movieData : movieObject});
     console.log(movieObject);

    }).catch((error) =>{
     console.log(error);
    });
  }

  fetchMovie(){
    fetch('http://www.omdbapi.com/?t=batman&apikey=25a585bd')
    .then(response => response.json())
    .then(movie => this.setState({ movie }));
  }

  render(){
    const movie = this.state.movie ? <h1>{this.state.movie.Title}</h1> : null;

    const {movies, movieData} = this.state;

    const movieInfo = this.state.movieData ? <MovieDetails movie={movieData} /> : null;
    const movieList = movies.map(movie => {
      return <Result movie={movie} key={movie.imdbID} receiver={this.handleMovieClickReceiver} />;
    });


    return (
      <div>
        <SearchForm
          receiver = {this.handleClickReceiver}
        />
        <div>
          {movieList}

      </div>

        <div>
          {movieInfo}
        </div>
      </div>


    )
  }
}

export default App;
