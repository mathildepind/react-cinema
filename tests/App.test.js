import React from 'react';
import App from '../src/App';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe ('App component', () => {
  it('starts with a movie', () => {
    const movies = [{
      Title : 'Batman Begins',
      Year : '2005',
      Poster : "https://images-na.ssl-images-amazon.com/images/M/MV5BYzc4ODgyZmYtMGFkZC00NGQyLWJiMDItMmFmNjJiZjcxYzVmXkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg"
    }, {
      Title : 'Superman Saves the World',
      Year : '1978',
      Poster : "https://images-na.ssl-images-amazon.com/images/M/MV5BYzc4ODgyZmYtMGFkZC00NGQyLWJiMDItMmFmNjJiZjcxYzVmXkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg"
      }
    ]

    const wrapper = shallow(<App movies={movies} />);
    const resultComponent = wrapper.find('movieList');
    console.log(resultComponent);
    expect(resultComponent.get(0).state.movie).toEqual({
      title : 'Batman Begins',
      year : '2005',
      poster : "https://images-na.ssl-images-amazon.com/images/M/MV5BYzc4ODgyZmYtMGFkZC00NGQyLWJiMDItMmFmNjJiZjcxYzVmXkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg"
    });
  });
});
