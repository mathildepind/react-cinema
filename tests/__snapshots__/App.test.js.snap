// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div />
  <div>
    <div>
      <h2 />
      <h3 />
      <h3 />
      <h4 />
      <img
        alt={undefined}
        src={undefined}
      />
    </div>
  </div>
</div>
`;
